### ISSUE TITLE
Error parsing jpeg image

### ISSUE THREAD

**Issue by lokedhs** [robert-strandh/McCLIM#167]
To reproduce, run the following function:

```
(defun test-error ()
  (let ((url "https://mamot.fr/system/accounts/avatars/000/012/095/original/c4c287a5f9d91a31.jpg?1491856580"))
    (multiple-value-bind (content code headers)
        (drakma:http-request url :force-binary t)
      (when (eql code 200)
        (let ((type (cdr (assoc :content-type headers))))
          (assert (equal type "image/jpeg"))
          (uiop:with-temporary-file (:stream stream :pathname file :type "jpg")
            (write-sequence content stream)
            :close-stream
            (clim:make-pattern-from-bitmap-file file)))))))
```

This will give the following error:

```
The value
  #2A((151 147 148 180 160 ...))
is not of type
  (ARRAY * (* * *))
```

Here is the stack trace when the error happens:

```
  0: (CLIM-INTERNALS::OPTICL-READ-BITMAP-FILE #P"/tmp/tmpX2JYJDQE-tmp.jpg")
  1: (CLIM:MAKE-PATTERN-FROM-BITMAP-FILE #P"/tmp/tmpX2JYJDQE-tmp.jpg" :DESIGNS NIL :FORMAT :BITMAP :PORT NIL)
  2: (UIOP/STREAM:CALL-WITH-TEMPORARY-FILE #<CLOSURE (FLET #:BEFORE150 :IN TEST-ERROR) {7FFFEE28DD2B}> :WANT-STREAM-P T :WANT-PATHNAME-P T :DIRECTION :IO :KEEP NIL :AFTER #<FUNCTION (FLET #:AFTER151 :IN TE..
  3: (TEST-ERROR)
```

**Comment by lokedhs** [comment]
This problem happens if the image is greyscale. Ten the image has only a single colour channel, and opticl will load the image data as a 2D array instead of 3D.

**Comment by dkochmanski** [comment]
fixed by pr #168
